src/online2bin/online2-wav-nnet3-latgen-grammar.cc

```diff
@@ -179,8 +179,8 @@ int main(int argc, char *argv[]) {
     nnet3::DecodableNnetSimpleLoopedInfo decodable_info(decodable_opts,
                                                         &am_nnet);
 
-
-    fst::GrammarFst fst;
+    // fst::GrammarFstTpl<const fst::ConstFst<fst::StdArc> > fst;
+    fst::GrammarFstTpl<StdVectorFst> fst;
```

Comment:
@danpovey I will revert this to use the ConstFst class (the default before my change). Do you think that kaldi needs to support multiple template classes of `GrammarFstTpl` here or anywhere? It would result in a lot of messy code. I'm leaning towards leaving all of the programs using ConstFst _only_ and adding mutability on an as-needed basis later on.

The one exception is `make-grammar-fst.cc`, where I added support for building mutable grammarfsts

Reply:
Yes please do.
Keeping it simple is the best thing, I think.  LMK when you're confident it's ready.